src/StaticLS/IDE/CodeActions/Parse.hs

```diff
@@ -0,0 +1,136 @@
+module StaticLS.IDE.CodeActions.Parse (
+  ActionableIssue (..),
+  Ignored (..),
+  KnownExtension (..),
+  actionableIssue,
+) where
+
+import Control.Applicative ((<|>))
+import Control.Monad (guard, (<=<))
+import Data.Foldable (asum, toList)
+import Data.Functor (($>))
+import Data.Text qualified as T
+import Language.Haskell.TH.LanguageExtensions (Extension)
+import Language.LSP.Protocol.Types qualified as LSP
+import Text.Regex.TDFA (getAllTextMatches, (=~), (=~~))
+
+-- This allows us to ignore diagnostics when comparing `ActionableIssue`s.
+-- Otherwise, we end up with duplicates with slightly different wording.
+newtype Ignored a = Ignored a deriving (Show)
+instance Eq (Ignored a) where (==) _ _ = True
+instance Ord (Ignored a) where compare _ _ = EQ
+
+data KnownExtension = KnownExtension {-# UNPACK #-} !Extension {-# UNPACK #-} !T.Text
+  deriving (Eq, Ord, Show)
+
+{-# NOINLINE knownExtensions #-}
+knownExtensions :: [KnownExtension]
+knownExtensions = (\ext -> KnownExtension ext $ T.pack $ show ext) <$> [minBound .. maxBound]
+
+data ActionableIssue
```

Comment:
An `ActionableIssue` is a defunctionalization of a method of creating one or more quick fixes. They are constructed by parsing GHC diagnostic messages, extracting the info relevant for making text edits in the relevant source file.